llvm: Get rid of warnings and add CMAKE_GDB

JaagupAverin opened the pull request:
* Reorganized clang/compiler_flags.cmake to match the order of gcc/compiler_flags.cmake
* Removed "-Wno-volatile" from clang flags as its not supported
* Exported CMAKE_GDB variable under target.cmake
* Added `__maybe_unused` to thread ID, as clang will otherwise genrate -Wunused-const-variable warnings

pdgendt commented:
@JaagupAverin you need to update your git `config.name` to match your full name to fix the CI issues.

```shell
$ git show HEAD
commit 71a478bfb8c32d20de49a14e216e3cd6221f50be (HEAD -> llvm-warnings)
Author: JaagupA <jaagup.averin@gmail.com>
```

JaagupAverin committed:
llvm: Reorganize clang/compiler_flags.cmake

Since this overrides values from gcc/compiler_flags.cmake,
they should ideally have the same structure and order.

Signed-off-by: Jaagup Averin <jaagup.averin@gmail.com>

JaagupAverin committed:
llvm: Remove "-Wno-volatile" from clang compiler args.

-Wno-volatile is set by gcc/compiler_flags.cmake,
but is not supported by clang and generated warnings.

Signed-off-by: Jaagup Averin <jaagup.averin@gmail.com>

JaagupAverin committed:
llvm: Get rid of -Wunused-const-variable warnings.

Often the thread k_tid_t value is not referenced for
self-standing threads, and should therefore be allowed
to remain unused.

Signed-off-by: Jaagup Averin <jaagup.averin@gmail.com>

JaagupAverin committed:
llvm: Export CMAKE_GDB during configuration.

Appears that llvm was forgotten when other toolchains
got their CMAKE_GDB export.

Signed-off-by: Jaagup Averin <jaagup.averin@gmail.com>
